assets/src/scripts/components/dailyValueHydrograph/selectors/time-series-data.js

```diff
@@ -0,0 +1,151 @@
+    (timeSeries) => {
+        if (!timeSeries) {
+            return [];
+        }
+
+        const timeStepInMillis = timeSeries.properties.timeStep.map((t) => new DateTime.fromISO(t, {zone: 'UTC'}).toMillis());
+
+        let lineSegments = [];
+        let previousDate = timeStepInMillis[0];
+        let previousApprovals = timeSeries.properties.approvals[0];
+        let segment = {
+            points: [],
+            approvals: timeSeries.properties.approvals[0]
+        };
+        timeStepInMillis.forEach(function(date, index) {
+            const resultValue = parseFloat(timeSeries.properties.result[index]);
+            const hasGap = date - previousDate >= TWO_DAYS;
+            const hasDifferentApprovals = !isEqual(timeSeries.properties.approvals[index], previousApprovals);
+            if (hasDifferentApprovals && !hasGap) {
+                // Add the current point to the last segment so that line is continuous
+                segment.points.push({
+                    value: resultValue,
+                    date: date
+                });
+            }
+
+            if (hasGap || hasDifferentApprovals) {
+                lineSegments.push(segment);
+                segment = {
+                    points: [],
+                    approvals: timeSeries.properties.approvals[index]
+                };
+                previousApprovals = timeSeries.properties.approvals[index];
+            }
+            segment.points.push({
+                value: parseFloat(resultValue),
+                date: date
+            });
+            previousDate = date;
+        });
+        lineSegments.push(segment);
+        return lineSegments;
+    }
+);
+
+/*
+ * Return a selector function that returns the epoch time for the current observations cursor offset.
+ * Return null if no current observations cursor offset is set.
+ */
+export const getCursorEpochTime = createSelector(
+    getObservationsCursorOffset,
+    getXScale,
+    (cursorOffset, xScale) => {
+
+        if (!cursorOffset) {
+            return xScale.domain()[1];
+        }
+        return xScale.domain()[0] + cursorOffset;
+    }
+);
+
+/*
+ * Return a selector which returns the points nearest the cursor's epoch time.
+ */
+export const getDataAtCursor = createSelector(
+    getCursorEpochTime,
+    getCurrentTimeSeriesPoints,
+    (cursorEpochTime, points)=> {
+        console.log('Cursor epoch time is ' + cursorEpochTime);
+        if (!points.length) {
+            return null;
+        }
+        console.log(`point at nearest time ${getNearestTime(points, cursorEpochTime).dateTime}`);
+        return getNearestTime(points, cursorEpochTime);
+    }
+);
+
+/*
+ * Return a selector which returns the x, y coordinates of the cursor
+ */
+export const getCursorPoint = createSelector(
+    getDataAtCursor,
+    getXScale,
+    getYScale,
+    (point, xScale, yScale) => {
+        if (!point) {
+            return [];
+        }
+        const result = [{
```

Comment:
Because this selector is used for a rendering function that is shared between the IV hydrograph and the DV hydrograph. In the IV hydrograph, you can have multiple points at the cursor offset. So I could wrap this selector where it is used but it seemed prudent to return an array since I bet this may more. I will update the comment to make it clear that it should be an array.